packages/cli/src/commands/doctor/doctor.js

```diff
@@ -0,0 +1,131 @@
+import chalk from 'chalk';
+import envinfo from 'envinfo';
+import {logger} from '@react-native-community/cli-tools';
+import ISSUES from './issues';
+import {getLoader} from '../../tools/loader';
+import printFixOptions, {KEYS} from './printFixOptions';
+import runAutomaticFix, {AUTOMATIC_FIX_LEVELS} from './runAutomaticFix';
+
+const printCategory = ({label, key}) => {
+  if (key > 0) {
+    logger.log();
+  }
+
+  logger.log(chalk.dim(label));
+};
+
+const printIssue = ({label, needsToBeFixed, isRequired}) => {
+  const symbol = needsToBeFixed
+    ? isRequired
+      ? chalk.red('✖')
+      : chalk.yellow('●')
+    : chalk.green('✓');
+
+  logger.log(` ${symbol} ${label}`);
+};
+
+export default (async function runDoctor() {
+  const Loader = getLoader();
+  const loader = new Loader();
+
+  loader.start('Running diagnostics...');
+
+  const environmentInfo = JSON.parse(
+    await envinfo.run(
+      {
+        Binaries: ['Node', 'Yarn', 'npm', 'Watchman'],
+        IDEs: ['Xcode', 'Android Studio'],
+        SDKs: ['iOS SDK', 'Android SDK'],
+        npmPackages: ['react', 'react-native', '@react-native-community/cli'],
+        npmGlobalPackages: ['*react-native*'],
+      },
+      {json: true, showNotFound: true},
+    ),
+  );
+
+  const iterateOverIssues = async ({label, issues}) => ({
+    label,
+    issues: (await Promise.all(
+      issues.map(async issue => {
+        if (issue.visible === false) {
+          return;
+        }
+
+        const {needsToBeFixed} = issue.getDiagnostics
+          ? issue.getDiagnostics(environmentInfo)
+          : await issue.getDiagnosticsAsync(environmentInfo);
+
+        return {
+          label: issue.label,
+          needsToBeFixed,
+          runAutomaticFix: issue.runAutomaticFix,
+          // Assume that it's required unless specified otherwise
+          isRequired:
+            typeof issue.isRequired === 'undefined' ? true : issue.isRequired,
+        };
+      }),
+    )).filter(issue => !!issue),
+  });
+
+  const iterateOverCategories = categories =>
+    Promise.all(categories.map(iterateOverIssues));
+
+  const issuesPerCategory = await iterateOverCategories(Object.values(ISSUES));
+
+  loader.stop();
```

Comment:
Note that the OraMock does not have a stop function, causing:

<img width="840" alt="Screenshot 2019-08-07 11 49 45" src="https://user-images.githubusercontent.com/2440089/62617297-74fb9600-b909-11e9-997e-97609e2dcae9.png">

Reply:
I wonder how you got this as I don't have it locally.

It does have a `stop` method: https://github.com/sindresorhus/ora#stop.

Reply:
I got it now by running with `--verbose`, I'll check that out, thanks!

Reply:
If you look at tools/loader.js the `getLoader` returns a `OraMock` if `verbose`. I guess we could just add a `stop` stub to `OraMock`